File: IT/Database/Esercizi/Python/esercizio7.py
# file esercizio7.py

from __future__ import print_function
import sqlite3

conn = sqlite3.connect("students.db")
cursor = conn.cursor()

try:
	cursor.execute("ALTER TABLE students ADD COLUMN grade INTEGER")
	conn.commit()
	print("Modificata tabella degli studenti.")

except sqlite3.OperationalError as e:
	print(e)

File: IT/2/Database/Esercizi/C++/esercizio7.cpp
/* file esercizio7.cpp */

#include <iostream>
#include <sqlite3.h>

using std::cout;
using std::endl;

int main()
{
	sqlite3 *db;
	sqlite3_open("students.db", &db);

	int rc = sqlite3_exec(db, "ALTER TABLE students ADD COLUMN grade INTEGER", NULL, NULL, NULL);

	if (rc != SQLITE_ERROR) {
		cout << "Modificato database studenti del corso di programmazione." << endl;
	}
	else {
		cout << sqlite3_errmsg(db) << endl;
	}

	sqlite3_close(db);
}

